### DISCUSSION TITLE
Could it be made simpler to unvendor third party dependencies?

### DISCUSSION THREAD

**Discussion by wolfv** [bulletphysics/bullet3#3368] [General]
I am currently packaging bullet for conda-forge, and unfortunately it comes with a couple of third-party dependencies that would be nice if they could be unvendored easily:

- lua
- eigen
- openvr
- gwen (discussed in #3073 that this is really part of bullet)
- zlib
- tinyxml2 (that is not installed, so that's OK)

I know that it's much easier to carry those in the tree, especially when supporting multiple operating systems (a problem we're actively trying to solve with conda-forge!).

Maybe if there was a CMake flag to use system / conda dependencies for the ones listed above that would be really cool. Then conda users would not have two copies of Eigen and other libraries in their installs.

I think Debian or Fedora would have similar issues when packaging up bullet, wouldn't they?

**Comment by erwincoumans** [comment]
Are you packaging bullet (for C++) or pybullet (for Python) or both?
Either way, you don't need lua (ignore/exclude the build3 folder), you don't need openvr either. It should be easy to replace anything in the examples/ThirdPartyLibs, such as Eigen, zlib and libxml2. The only dependency you need to include is Gwen. We can update the CMakeLists.txt to make that easier.
